src/git.rs

```diff
@@ -97,13 +97,18 @@ impl GitWrapper for Git {
         self.run_cmd(remote_update_cmd)
     }
 
-    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf) -> Result {
+    fn git_push_mirror(&self, dest: &str, repo_dir: &PathBuf, refspec: &Option<String>) -> Result {
         let mut push_cmd = self.git_base_cmd();
-        push_cmd
-            .current_dir(repo_dir)
-            .args(&["push", "--mirror"])
-            .arg(dest);
-
+        push_cmd.current_dir(repo_dir);
+        push_cmd.arg("push");
+        match &refspec {
+            Some(r) => {
+                push_cmd.args(&[dest, format!("{0}:refs/heads/{0}", r).as_str()]);
```

Comment:
How can I push a tag with this? I'm still not entirely happy with this. I would likt to be able to do something like: `refs/pull/*:+refs/heads/pull/*` here.